bindings/wasm/examples/editor.html

```diff
@@ -129,6 +129,42 @@
   var Module = {
     onRuntimeInitialized: function () {
       Module.setup();
+      // Setup memory management, such that users don't have to care about
+      // calling `delete` manually.
+      // Note that this only fixes memory leak across different runs: the memory
+      // will only be freed when the compilation finishes.
+
+      let manifoldRegistry = [];
+      for (const name of
+               ['add', 'subtract', 'intersect', 'refine', 'transform', 'translate',
+                'rotate', 'scale']) {
+        const originalFn = Module.Manifold.prototype[name];
+        Module.Manifold.prototype["_" + name] = originalFn;
+        Module.Manifold.prototype[name] = function(...args) {
+          const result = this["_" + name](...args);
+          manifoldRegistry.push(result);
+          return result;
+        }
+      }
+
+      for (const name
+               of ['cube', 'cylinder', 'sphere', 'extrude', 'revolve', 'union',
+                   'difference', 'intersection']) {
+        const originalFn = Module[name];
+        Module[name] = function(...args) {
+          const result = originalFn(...args);
+          manifoldRegistry.push(result);
+          return result;
+        }
+      }
+
+      Module.cleanup = function() {
+        for (const obj of manifoldRegistry) {
+          obj.delete();
+        }
+        manifoldRegistry = [];
```

Comment:
This looks great. Should we build this into the JS bindings themselves instead? Seems like anyone using this library would benefit from a cleanup utility. Or does it make assumptions that aren't necessarily valid? Anyway, this is great for now, but food for thought.

Reply:
This assumes that the user will not call `delete` manually for the objects. For the editor, we did not add the delete method to typescript definition and did not document about it, so I assume that users will not call delete on it.

Reply:
Yeah, that sounds good. Still, there is the issue of large chains of boolean operations potentially needing to clean up memory intermittently to keep from running out before completion. We may want to think through how to do that before exposing a memory-management API.